src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+package com.microsoft.partnercatalyst.fortis.spark
+
+import java.time.Duration
+import java.util.concurrent.locks.ReentrantLock
+import java.util.concurrent.{CompletableFuture, SynchronousQueue, TimeUnit}
+
+import com.microsoft.azure.servicebus.primitives.ConnectionStringBuilder
+import com.microsoft.azure.servicebus._
+import com.microsoft.partnercatalyst.fortis.spark.TransformManager._
+import com.microsoft.partnercatalyst.fortis.spark.dba.ConfigurationManager
+import com.microsoft.partnercatalyst.fortis.spark.dto.{BlacklistedTerm, SiteSettings}
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import com.microsoft.partnercatalyst.fortis.spark.transforms.image.{ImageAnalysisAuth, ImageAnalyzer}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.language.{LanguageDetector, LanguageDetectorAuth}
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.LocationsExtractorFactory
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client.FeatureServiceClient
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetectorAuth
+import org.apache.spark.SparkContext
+
+import scala.util.Properties
+
+class TransformManager(configManager: ConfigurationManager, featureServiceClient: FeatureServiceClient) extends Serializable with Loggable {
+  private val shouldUpdate: SynchronousQueue[Delta] = new SynchronousQueue[Delta]()
+  private val writeLock: ReentrantLock = new ReentrantLock(true)
+
+  // Do not serialize these values. The transformContext would otherwise contain stale Broadcast
+  // instances, so we rebuild it on recovery from checkpoint.
+  @volatile @transient private var transformContext: TransformContext = null
+  @volatile @transient private var queueClient: QueueClient = null
+
+  def getOrUpdateContext(sparkContext: SparkContext): TransformContext = {
+    ensureInitialized(sparkContext)
+
+    writeLock.lock()
+    try {
+      // Grab delta from hand-off only if it's available *now*
```

Comment:
This comment tells me what the code below is doing, but not why. It would be very useful to know why it's important that the hand-off is available right now.